abci/example/kvstore/kvstore.go

```diff
@@ -86,14 +86,13 @@ func (app *Application) Info(req types.RequestInfo) (resInfo types.ResponseInfo)
 }
 
 // tx is either "key=value" or just arbitrary bytes
-func (app *Application) DeliverTx(req types.RequestDeliverTx) types.ResponseDeliverTx {
+func (app *Application) DeliverTx(tx []byte) *types.ResponseDeliverTx {
```

Comment:
> The problem is that the design pattern used here (having an app contain another app and only implement extra functionality) is not a good fit for FinalizeBlock

I'm not totally understand where that pattern is being used. It looks like in the applications `FinalizeBlock`, it just calls this `DeliverTx` in a loop. We could just just either just change this to be an unexported helper method, preferably with a different name, or move the actual logic into `FinalizeBlock`. The reason I'd advocate for getting rid of `DeliverTx` is that leaving it around will make it a little unclear to readers of this code what the actual status of `DeliverTx` is. Removing references to it means that it will be clear to future readers of the code that it no longer exists and we can avoid confusion / ambiguity.

Reply:
Let's talk 1:1 will be more efficient

Reply:
Thanks @williambanfield  for the discussion. We agreed to rename `DeliverTx` here to `HandleTx` to avoid any confusion with the old API, for anyone that might come and read this code later on.